aptos-move/move-examples/defi/sources/locked_coins.move

```diff
@@ -0,0 +1,145 @@
+module defi::locked_coins {
+    use aptos_framework::account;
+    use aptos_framework::coin::{Self, Coin};
+    use aptos_framework::event::{Self, EventHandle};
+    use aptos_framework::timestamp;
+    use aptos_std::table::{Self, Table};
+    use std::error;
+    use std::signer;
+
+    /// Represents a lock of coins until some specified unlock time. Afterward, the recipient can claim the coins.
+    struct Lock<phantom CoinType> has store {
+        coins: Coin<CoinType>,
+        unlock_time_secs: u64,
+    }
+
+    /// Holder for a map from recipients => locks.
+    /// There can be at most one lock per recipient.
+    struct Locks<phantom CoinType> has key {
+        locks: Table<address, Lock<CoinType>>,
```

Comment:
what happens if different amount of coined with different lock time per address?

Reply:
That's not a problem. In fact, it's how this is meant to work :) You can give different amounts with different lock durations to different accounts